packages/logger/src/Logger.ts

```diff
@@ -240,7 +238,7 @@ class Logger extends Utility implements ClassThatLogs {
       logFormatter: this.getLogFormatter(),
     };
     const parentsPowertoolsLogData = this.getPowertoolLogData();
-    const childLogger = this.createLogger(
```

Comment:
If possible, we should try to keep this method as it allows people to more easily extend the logger. 

If we instantiate the logger as `const childLogger = new Logger()`, even if I create my own logger super class (i.e. `class MyOwnLogger extends Logger {}`), its children will always be of type `Logger` (the original).

While we don't strictly need it, by providing this method, people creating `MyOwnLogger` can override it and make it return an instance of `MyOwnLogger` like so:

```ts
class MyOwnLogger extends Logger {
  protected createLogger(options?: ConstructorOptions): MyOwnLogger {
    return new MyOwnLogger(options);
  }
}
```

More context here, where the issue was [discussed](https://github.com/aws-powertools/powertools-lambda-typescript/issues/1307#issuecomment-1625421828) & added.

Reply:
Thanks for the explanation. I'll reverse this change, maybe it make sense to add a link to this issue in the method comment section as a memo.

Reply:
I think you're right that we should document/explain this in the comment.

I'd rather though write a comment that explains how to use the method and what it's for rather than linking the issue.

We can reuse and rephrase the explanation/example I mentioned in my previous comment.

Reply:
Based on the discussion and your comment, I reversed this change with a detailed comment.